### PR TITLE
feat: Support running

### PR DESCRIPTION
**Pull request by gaocegege**
Signed-off-by: Ce Gao <gaoce@caicloud.io>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/894)

### CODE REVIEW

**Comment by gaocegege** [comment]
/hold

Until 0.7 is released

**Comment by johnugeorge** [comment]
@gaocegege Can you rebase? I will review it today

File: pkg/job/v1alpha3/job/job.go [hunk 1]
```diff
@@ -28,9 +28,10 @@ func (j Job) GetDeployedJobStatus(
 			log.Error(unerr, "NestedFieldCopy unstructured to status error")
 			return nil, unerr
 		}
-		log.Info("NestedFieldCopy unstructured to status error",
-			"err", "Status is not found in job")
-		return &jobCondition, nil
+		// Job does not have the running condition in status, thus we think
+		// the job is running when it is created.
```
Comment (johnugeorge) [inline pkg/job/v1alpha3/job/job.go]: Can you explain this comment? This is little confusing. When there is no status field, it means that job is just created.

File: pkg/job/v1alpha3/job/job.go [hunk 2]
```diff
@@ -28,9 +28,10 @@ func (j Job) GetDeployedJobStatus(
...
+		log.Info("NestedFieldCopy", "err", "status cannot be found in job")
```
Comment (johnugeorge) [inline pkg/job/v1alpha3/job/job.go]: Also, why is this logic only added for batch job and not for other types?

Comment (johnugeorge) [reply pkg/job/v1alpha3/job/job.go]: I meant, kubeflow provider

File: pkg/job/v1alpha3/job/job.go (hunk 1 above)
Comment (gaocegege) [reply pkg/job/v1alpha3/job/job.go]: Yeah, The Job does not have a running condition, thus we cannot use the same logic as TFJob

File: pkg/job/v1alpha3/job/job.go (hunk 2 above)
Comment (gaocegege) [reply pkg/job/v1alpha3/job/job.go]: The kubeflow provider already implements it before.

**Comment by gaocegege** [comment]
/hold

Found some problems after rebase.

**Comment by gaocegege** [comment]
Verified with job:

```json
(base) ➜  katib git:(running) ✗ kubectl -n kubeflow get trials -o json | jq ".items[] | {conditions: .status.conditions}"
{
  "conditions": [
    {
      "lastTransitionTime": "2019-12-05T02:26:59Z",
      "lastUpdateTime": "2019-12-05T02:26:59Z",
      "message": "Trial is created",
      "reason": "TrialCreated",
      "status": "True",
      "type": "Created"
    },
    {
      "lastTransitionTime": "2019-12-05T02:29:27Z",
      "lastUpdateTime": "2019-12-05T02:29:27Z",
      "message": "Trial is running",
      "reason": "TrialRunning",
      "status": "False",
      "type": "Running"
    },
    {
      "lastTransitionTime": "2019-12-05T02:29:27Z",
      "lastUpdateTime": "2019-12-05T02:29:27Z",
      "message": "Trial has succeeded",
      "reason": "TrialSucceeded",
      "status": "True",
      "type": "Succeeded"
    }
  ]
}
{
  "conditions": [
    {
      "lastTransitionTime": "2019-12-05T02:26:42Z",
      "lastUpdateTime": "2019-12-05T02:26:42Z",
      "message": "Trial is created",
      "reason": "TrialCreated",
      "status": "True",
      "type": "Created"
    },
    {
      "lastTransitionTime": "2019-12-05T02:26:43Z",
      "lastUpdateTime": "2019-12-05T02:26:43Z",
      "message": "Trial is running",
      "reason": "TrialRunning",
      "status": "True",
      "type": "Running"
    }
  ]
}
{
  "conditions": [
    {
      "lastTransitionTime": "2019-12-05T02:26:42Z",
      "lastUpdateTime": "2019-12-05T02:26:42Z",
      "message": "Trial is created",
      "reason": "TrialCreated",
      "status": "True",
      "type": "Created"
    },
    {
      "lastTransitionTime": "2019-12-05T02:26:59Z",
      "lastUpdateTime": "2019-12-05T02:26:59Z",
      "message": "Trial is running",
      "reason": "TrialRunning",
      "status": "True",
      "type": "Running"
    }
  ]
}
```

File: pkg/job/v1alpha3/job/job.go (hunk 2 above)
Comment (johnugeorge) [reply pkg/job/v1alpha3/job/job.go]: https://github.com/kubeflow/katib/blob/master/pkg/job/v1alpha3/kubeflow/kubeflow.go#L39

**Comment by gaocegege** [comment]
/hold 

Forgot to test on TFJob/PyTorchJob

File: pkg/controller.v1alpha3/trial/trial_controller_util.go [hunk 3]
```diff
@@ -118,6 +130,9 @@ func (r *ReconcileTrial) updateFinalizers(instance *trialsv1alpha3.Trial, finali
 }
 
 func isTrialObservationAvailable(instance *trialsv1alpha3.Trial) bool {
+	if instance == nil {
+		return false
+	}
```
Comment (johnugeorge) [inline pkg/controller.v1alpha3/trial/trial_controller_util.go]: just wondering when instance becomes nil ?

Comment (gaocegege) [reply pkg/controller.v1alpha3/trial/trial_controller_util.go]: Never, but prefer to keep to avoid potential crashes.

**Pull request merged**